core/src/main/java/org/apache/calcite/rel/rules/ProjectWindowTransposeRule.java

```diff
@@ -99,7 +100,7 @@ public ProjectWindowTransposeRule(RelBuilderFactory relBuilderFactory) {
 
     final LogicalProject projectBelowWindow =
         new LogicalProject(cluster, window.getTraitSet(), ImmutableList.of(),
-            window.getInput(), exps, builder.build());
+            window.getInput(), exps, builder.build(), ImmutableSet.of());
```

Comment:
If my understanding correct, we only transpose an identity project, otherwise the identity project will be derived and passed below the window and the topmost project will be adjusted. In that case it looks safe.

> Also it seems that here we are going to lose the variables set in the original projection

It's a bug, thank you for noticing